app/components/custom_status/custom_status_expiry.tsx

```diff
@@ -0,0 +1,146 @@
+
+    const plusSixDaysEndTime = currentMomentTime.clone().add(6, 'days').endOf('day');
+    const tomorrowEndTime = currentMomentTime.clone().add(1, 'day').endOf('day');
+    const todayEndTime = currentMomentTime.clone().endOf('day');
+    const isCurrentYear = currentMomentTime.get('y') === expiryMomentTime.get('y');
+
+    let format = '';
+    let renderDayOrDate;
+
+    if (expiryMomentTime.isSame(todayEndTime)) {
+        renderDayOrDate = (
+            <FormattedText
+                id='custom_status.expiry_time.today'
+                defaultMessage='Today'
+            />
+        );
+    } else if (expiryMomentTime.isAfter(todayEndTime) && expiryMomentTime.isSameOrBefore(tomorrowEndTime)) {
+        renderDayOrDate = (
+            <Text>
+                <FormattedText
+                    id='custom_status.expiry_time.tomorrow'
+                    defaultMessage='Tomorrow'
+                />
+                {' '}
+                <FormattedText
+                    id='custom_status.expiry.at'
+                    defaultMessage='at'
+                />
+                {' '}
+            </Text>
+        );
+    }
+
+    const useDay = expiryMomentTime.isBetween(tomorrowEndTime, plusSixDaysEndTime);
+    const useDate = expiryMomentTime.isAfter(plusSixDaysEndTime) && isCurrentYear;
+
+    if (useDay) {
+        format = 'dddd';
+    } else if (useDate) {
+        format = 'MMM DD';
+    } else if (!isCurrentYear) {
+        format = 'MMM DD, YYYY';
+    }
+
+    if (expiryMomentTime.isAfter(tomorrowEndTime)) {
+        renderDayOrDate = (
+            <FormattedDate
+                format={format}
+                timezone={timezone}
+                value={expiryMomentTime.toDate()}
+            />
+        );
+
+        renderDayOrDate = showTimeCompulsory ? (
+            <Text>
+                {renderDayOrDate}
+                {' '}
+                <FormattedText
+                    id='custom_status.expiry.at'
+                    defaultMessage='at'
+                />
+                {' '}
+            </Text>
+        ) : renderDayOrDate;
+    }
+
+    const useTime = showTimeCompulsory || !(expiryMomentTime.isSame(todayEndTime) || expiryMomentTime.isAfter(tomorrowEndTime));
+
+    const renderTime = useTime && (
+        <FormattedTime
+            isMilitaryTime={militaryTime}
+            timezone={timezone || ''}
+            value={expiryMomentTime.toDate()}
+        />
+    );
+
+    const prefix = showPrefix && (
+        <Text>
+            <FormattedText
+                id='custom_status.expiry.until'
+                defaultMessage='Until'
+            />{' '}
+        </Text>
+    );
+
+    return (
+        <Text
+            testID={testID}
+            style={[styles.text, textStyles]}
+        >
+            {withinBrackets && '('}
+            {prefix}
+            {renderDayOrDate}
+            {renderTime}
+            {withinBrackets && ')'}
+        </Text>
+    );
+};
```

Comment:
@enahum are we sure we want to create variables for simple values here? IMO it would be more effective to just insert it in the return value, WDYT?